apps/passport-client/components/modals/ResolveSubscriptionError.tsx

```diff
@@ -71,30 +67,31 @@ function FetchError({
   const [polling, setPolling] = useState<boolean>(false);
   const [stillFailing, setStillFailing] = useState<boolean>(false);
   const [error, setError] = useState<SubscriptionFetchError>();
-  const credentialCache = useCredentialCache();
-
-  const identity = useIdentity();
-  const pcds = usePCDCollection();
+  const dispatch = useDispatch();
 
   const onRefreshClick = useCallback(async () => {
     setPolling(true);
 
-    const credentialManager = new CredentialManager(
-      identity,
-      pcds,
-      credentialCache
-    );
-    await subscriptions.pollSingleSubscription(subscription, credentialManager);
-    setPolling(false);
-    const error = subscriptions.getError(subscription.id);
-    if (error && error.type === SubscriptionErrorType.FetchError) {
-      setStillFailing(true);
-      setError(error);
-    } else {
-      setStillFailing(false);
-      setError(undefined);
-    }
-  }, [identity, pcds, credentialCache, subscriptions, subscription]);
+    dispatch({
+      type: "sync-subscription",
+      subscriptionId: subscription.id,
+      onSuccess: () => {
+        setPolling(false);
+        const error = subscriptions.getError(subscription.id);
+        if (error && error.type === SubscriptionErrorType.FetchError) {
```

Comment:
The way this dispatch is handling errors is very odd.  E.g. the fact that a subscription fetch error is treated as success and needs to be checked here.  I think the dispatch handler should really be checking this, and not trying to apply actions if the fetch failed.  That being said, though, this isn't a behavior change in this PR.

I also worry about errors which aren't of type FetchError (which get treated as stillFailing=false below).  That may not be a valid situation today (I didn't look at whether there are non-fetch errors), but I feel like the code below should be based on the value of `error` rather than assuming it's undefined.  Again, not a behavior change in this PR, though.